### PR TITLE
Fixed #221, where resolving generic method arguments which are inner classes fails.

### LINKED ISSUE #221: Failing to resolve generic method arguments if they are inner static generic classes
There seem to be several issue causing this:

1.  `JavassistTypeDeclarationAdapter#containerType()` throws when in case of an inner class:

     ```java
      if (ctClass.getDeclaringClass() == null) {
        return Optional.empty();
      } else { // this is an inner class
        throw new UnsupportedOperationException();
      }
    ```
2. `JavassistUtils#objectTypeArgumentToType(...)`  calls `typeSolver.solveType(typeName)`, however, if `typeName` is an inner class coming from a `JarTypeSolver`, it may contain `$` characters, which mess up downstream resolution so I guess they need to be replaced with plain old dots (`.`).

3.  `JavassistUtils#signatureTypeToType(...)` calls `typeSolver.solveType(classType.getName())` however, if `classType` happens to be an inner class, the `getName()` method returns only the last name fragment, dropping the prefix (i.e., `ClassX.ClassY.ClassZ` => `ClassZ`). 
The short name is then passed to the type resolver likes so `typeSolver.solveType(classType.getName())` which fails due to the name being unqualified.

While I have a reproducible use-case for this issue (based on which have submitted #222), I'm afraid I was not able to synthesise it so that it has a proper unit test, mostly on account of my local setup having numerous external jars form which things get resolved.

### PR DESCRIPTION
**Pull request by staslev**
See #221.

### CODE REVIEW

File: java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java [hunk 1]
```diff
@@ -122,21 +122,27 @@ static Type signatureTypeToType(SignatureAttribute.Type signatureType, TypeSolve
...
+                classType.getDeclaringClass().getName() + "." + classType.getName() :
+                classType.getName();
+            ReferenceTypeDeclaration typeDeclaration = typeSolver.solveType(resolveFriendly(typeName));
             return new ReferenceTypeImpl(typeDeclaration, typeParameters, typeSolver);
         } else {
             throw new RuntimeException(signatureType.getClass().getCanonicalName());
         }
     }
 
+    private static String resolveFriendly(String typeName) {
```
Comment (ftomassetti) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java]: Can you rename and comment this method?

**Comment by ftomassetti** [comment]
Aside from my comment, this looks good. You are on a roll, great!

File: java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java (hunk 1 above)
Comment (staslev) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java]: Sure, what name did you have in mind?

Comment (ftomassetti) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java]: What about `internalNameToCanonicalName`?

Comment (staslev) [inline java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java]: Sounds good.

**Comment by staslev** [comment]
retest this please.

**Comment by ftomassetti** [comment]
Build restarted

**Comment by staslev** [comment]
Thanks!

**Comment by ftomassetti** [comment]
Passed, merging

**Pull request merged**